Library/Formula/nvm.rb

```diff
@@ -4,21 +4,42 @@ class Nvm < Formula
   sha256 "0008754e7421c55e66f36c4a83ca775928f7b2e7a0543098f0050d9780419b2a"
   head "https://github.com/creationix/nvm.git"
 
+  resource "nvm-exec" do
+    url "https://raw.githubusercontent.com/creationix/nvm/v0.24.1/nvm-exec"
+    sha256 "915118c666c6f3e03ce4a4545038ecae691c7c496579c614f4d32200a180924c"
+  end
+
   def install
     prefix.install "nvm.sh"
     bash_completion.install "bash_completion" => "nvm"
+
+    resource("nvm-exec").stage do
+      prefix.install("nvm-exec")
+      chmod 0755, "#{prefix}/nvm-exec"
+    end
   end
 
   def caveats; <<-EOS.undent
+      Add the following to $HOME/.bashrc, $HOME/.zshrc, or your shell's
+      equivalent configuration file:
+
+        source $(brew --prefix nvm)/nvm.sh
+
+
+      Optionally, if you prefer to have NVM's working directory in your $HOME
+      directory, do the following:
+
       Add NVM's working directory to your $HOME path (if it doesn't exist):
 
         mkdir ~/.nvm
 
-      Add the following to $HOME/.bashrc, $HOME/.zshrc, or your shell's
-      equivalent configuration file:
+      Copy nvm-exec to NVM's working directory
+
+        cp $(brew --prefix nvm)/nvm-exec ~/.nvm/
+
+      Add the following to your shell's configuration file:
 
         export NVM_DIR=~/.nvm
-        source $(brew --prefix nvm)/nvm.sh
 
       Type `nvm help` for further information.
```

Comment:
Can we slim these caveats a lot? There's a lot of text here and it feels like it's not all Homebrew-specific (only Homebrew-specific bits should be here).